core/src/main/java/services/servicesImpl/POISeviceImpl.java

```diff
@@ -0,0 +1,48 @@
+package services.servicesImpl;
+
+import entities.HeatMap;
+import entities.HeatPoint;
+import entities.POI;
+import repository.POIRepository;
+import services.POIService;
+import services.POISourceService;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class POISeviceImpl implements POIService{
+
+    public static final double cityX1 = 23.7;
+    public static final double cityX2 = 24.0;
+    public static final double cityY1 = 53.5;
+    public static final double cityY2 = 53.7;
+
+    private final POIRepository repository;
+
+    private final POISourceService sourceService;
+
+    public POISeviceImpl(POIRepository repository, POISourceService sourceService) {
+        this.repository = repository;
+        this.sourceService = sourceService;
+    }
+
+    public void loadAndStore() {
+       List<POI> list = sourceService.buildQuery(cityX1, cityY1, cityX2, cityY2);
+       repository.save(list);
+
+    }
+
+    public HeatMap build() {
+        HeatMap map = HeatMap.coreBuild();
+        for (HeatPoint point : map.getMap()){
+            double[] crd = point.getSquareByRadius();
+            List<POI> list = repository.getByBounds(crd[0], crd[1], crd[2], crd[3]);
+            Map<String, Object> attributes = new HashMap<String, Object>();
```

Comment:
Method can be refactored - no need to create map each time here - please check my comment about addAttribute above